app/assets/javascripts/components/warn-before-unload.js

```diff
@@ -0,0 +1,27 @@
+function WarnBeforeUnload ($module, $scope) {
+  if (!$module) $module = window
+  if (!$scope) $scope = document
+  this.$module = $module
+  this.$forms = $scope.querySelectorAll('[data-warn-before-unload=true]')
+}
+
+WarnBeforeUnload.prototype.init = function () {
+  var _this = this
+  this.$forms.forEach(function ($form) {
```

Comment:
This may result in a `Cannot read property 'forEach' of undefined`. Can we check for this.$forms to exist and return from this function if not?

Reply:
updated